src/t8_cmesh/t8_cmesh_testcases.c

```diff
@@ -311,20 +309,15 @@ t8_test_create_new_hypercube_cmesh (int cmesh_id)
      t8_eclass_to_string[eclass], t8_comm_string_list[comm_num], do_bcast,
      do_partition, periodic);
 
-  if (eclass == T8_ECLASS_PYRAMID) {
-    t8_debugf
-      ("Pyramids are not implemented (in this branch), therefore the eclass is changed to T8_ECLASS_HEX");
-    return t8_cmesh_new_hypercube (T8_ECLASS_HEX, comm, do_bcast,
+  if (periodic == 1) {
+    return t8_cmesh_new_hypercube (eclass, comm, do_bcast,
                                    do_partition, 0);
   }
+  else{
+      return t8_cmesh_new_hypercube (eclass, comm, do_bcast, do_partition,
+                                     periodic);
+  }
```

Comment:
It looks to me as this would always execute `t8_cmesh_new_hypercube (eclass, comm, do_bcast, do_partition, 0);`

- What is the purpose of this if?
- Do we call `t8_cmesh_new_hypercube (eclass, comm, do_bcast, do_partition, 0);` twice? If so, why?

Reply:
Updated it. It now avoids the case, that a periodic hypercube with pyramids is built.